src/types/interfaces.ts

```diff
@@ -205,45 +205,79 @@ export interface IConversationParticipant {
   confined: boolean;
 }
 
-export interface IJingleSession extends WildEmitter {
+export enum SessionEvents {
+  participantsUpdate = 'participantsUpdate',
+  activeVideoParticipantsUpdate = 'activeVideoParticipantsUpdate',
+  speakersUpdate = 'speakersUpdate',
+  incomingMedia = 'incomingMedia',
+  pinnedParticipant = 'pinnedParticipant'
+}
+
+export interface IExtendedMediaSession extends GenesysCloudMediaSession {
   id: string;
-  sid: string;
-  peerID: string;
   originalRoomJid: string;
   conversationId: string;
-  active: boolean;
   sessionType: SessionTypes;
-  streams: MediaStream[];
-  tracks: MediaStreamTrack[];
-  accept: () => void;
-  end: () => void;
-  addTrack: (track: MediaStreamTrack) => Promise<void>;
-  addStream: (stream: MediaStream) => Promise<void>;
-  removeTrack: (track: MediaStreamTrack) => Promise<void>;
-  replaceTrack: (newTrack: MediaStreamTrack, oldTrack?: MediaStreamTrack) => Promise<void>;
-  mute: (userId: string, mediaType: 'video' | 'audio') => void;
-  unmute: (userId: string, mediaType: 'video' | 'audio') => void;
-  pc: {
-    getSenders: () => RTCRtpSender[],
-    getReceivers: () => RTCRtpReceiver[],
-    pc: RTCPeerConnection
-  };
-  pcParticipant?: IConversationParticipant;
-  fromUserId?: string;
+  active: boolean;
   videoMuted?: boolean;
   audioMuted?: boolean;
+  fromUserId?: string;
+  pcParticipant?: IConversationParticipant;
   startScreenShare?: () => Promise<void>;
   stopScreenShare?: () => Promise<void>;
   pinParticipantVideo?: (participantId: string) => Promise<void>;
   _resurrectVideoOnScreenShareEnd?: boolean;
   _outboundStream?: MediaStream;
   _screenShareStream?: MediaStream;
   _outputAudioElement?: HTMLAudioElement & { sinkId?: string; setSinkId?: (deviceId: string) => Promise<any>; };
-  _statsGatherer?: WebrtcStatsGatherer;
   _lastParticipantsUpdate?: IParticipantsUpdate;
   _lastOnScreenUpdate?: IOnScreenParticipantsUpdate;
+
+  on: (event: MediaSessionEvents | SessionEvents, listener: (...data: any) => void) => void;
+  once: (event: MediaSessionEvents | SessionEvents, listener: (...data: any) => void) => void;
+  off: (event: MediaSessionEvents | SessionEvents, listener: (...data: any) => void) => void;
+  emit: (event: MediaSessionEvents | SessionEvents, ...data: any) => void;
```

Comment:
Does adding these types here cause typescript to get upset if you try to wildcard the events like `session.on('*', ...);`?

Reply:
`session.on('*'...)` isn't valid for event emitter.